src/services/tweets.service.ts

```diff
@@ -13,18 +14,60 @@ import {
 import { deleteLikes } from "./likes.service";
 import { updateTagsFromContent } from "./tags.service";
 
-export const getAllTweetsOfCurrentUser = (req: Request) => {
+export const getAllTweets = async (req: Request) => {
   const [query, options] = new PaginationParameters({ query: req.query }).get();
-  return getList({ ...query, author: req.user._id }, options);
+  options.leanWithId = true;
+  options.lean = true;
+  const list = await getList(query, options);
+  const listOfTweetId = list.docs.map((doc) => doc.id);
+  console.warn(listOfTweetId);
+  const liked = await getAll(
+    { user: req.user._id, tweet: { $in: listOfTweetId } },
+    {}
+  );
+
+  liked.docs.forEach((record) => {
+    listOfTweetId.forEach((currentId, index) => {
+      if (currentId == record.tweet._id) {
+        list.docs[index].isLiked = true;
+      }
+    });
+  });
+
+  return list;
 };
 
-export const getAllTweets = (req: Request) => {
+export const getAllTweetsOfCurrentUser = async (req: Request) => {
   const [query, options] = new PaginationParameters({ query: req.query }).get();
-  return getList(query, options);
+  options.leanWithId = true;
+  options.lean = true;
+
+  const list = await getList({ ...query, author: req.user._id }, options);
+  const listOfTweetId = list.docs.map((doc) => doc.id);
+  console.warn(listOfTweetId);
+  const liked = await getAll(
+    { user: req.user._id, tweet: { $in: listOfTweetId } },
+    {}
+  );
+
+  liked.docs.forEach((record) => {
+    listOfTweetId.forEach((currentId, index) => {
+      if (currentId == record.tweet._id) {
+        list.docs[index].isLiked = true;
+      }
+    });
+  });
+
+  return list;
```

Comment:
```suggestion

  return getAllTweets(req, req.user._id);
```
1) you can call function getAllTweets here, 
2) clean all other duplicate code
3) add not required params forUserId for function getAllTweets
4) add logic to getAllTweets to check filled param  forUserId or no

it is very bad to extend two identical functions